src/constants/actions.js

```diff
@@ -0,0 +1,19 @@
+module.exports = {
+  // action types
+  TRANSITION: 'ACTION_TYPE/TRANSITION',
+  METADATA: 'ACTION_TYPE/METADATA',
+  PROGRESS: 'ACTION_TYPE/PROGRESS',
+  // transition actions
+  REJECT: 'TRANSITION/REJECT',
```

Comment:
Yes, I think before and after hooks could work. Everything could be checked in the hooks w/o creating new routings.